### PR TITLE
reference/sql: update compatibility description of MySQL

### PR DESCRIPTION
**Pull request by tangenta**
### What is changed, added or deleted? 
Update reference/sql/mysql-compatibility.md.

### What is the related PR or file link(s)? 

- [ ] Reference link(s):
- [ ] This PR is to align with:
- [x] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [ ] dev (the latest development version)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [x] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [x] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

File: v3.0/reference/mysql-compatibility.md [hunk 1]
```diff
@@ -196,3 +210,12 @@ TiDB 不需要导入时区表数据也能使用所有时区名称，采用系统
...
 目前 TiDB 尚不能完整支持月为 0 或日为 0（但年不为 0）的日期。在非严格模式下，此类日期时间能被正常插入，但对于特定类型 SQL 可能出现无法读出来的情况。
+
+### 类型系统的区别
+
+以下的列类型 MySQL 支持，但 TiDB 不支持：
+
++ FLOAT4/FLOAT8
++ FIXED (alias for DECIMAL)
++ SERIAL (alias for BIGINT UNSIGNED NOT NULL AUTO_INCREMENT UNIQUE)
++ SQL_TSI_* (包括 SQL_TSI_YEAR，SQL_TSI_MONTH， SQL_TSI_WEEK， SQL_TSI_DAY， SQL_TSI_HOUR， SQL_TSI_MINUTE， SQL_TSI_SECOND)
```
Comment (yikeke) [inline v3.0/reference/mysql-compatibility.md]:
```suggestion
+ SQL_TSI_*（包括 SQL_TSI_YEAR、SQL_TSI_MONTH、SQL_TSI_WEEK、SQL_TSI_DAY、SQL_TSI_HOUR、SQL_TSI_MINUTE 和 SQL_TSI_SECOND）
```
格式参考规范请参见《技术文档风格指南》[“中英文混排格式”一节](https://docs.google.com/document/d/1b6ZhZD33OoM8AacpKksGGSuxJWReLkNnSt8eSc1kTXc/edit?ts=5c9c682e#heading=h.2w47pgl5ww0i)。

File: v3.0/reference/mysql-compatibility.md [hunk 2]
```diff
@@ -99,6 +100,19 @@ TiDB 支持常用的 MySQL 内建函数，但是不是所有的函数都已经
...
+    - `CHECK/DROP CHECK`
+    - `STATS_AUTO_RECALC/STATS_SAMPLE_PAGES`
+    - `SECONDARY_ENGINE`
+    - `ENCRYPTION`
++ Table Partition 不支持以下语法
+    - `PARTITION BY LIST`
+    - `PARTITION BY KEY`
+    - `SUBPARTITION`
+    - `{CHECK|EXCHANGE|TRUNCATE|OPTIMIZE|REPAIR|IMPORT|DISCARD|REBUILD|REORGANIZE} PARTITION`
+
```
Comment (yikeke) [inline v3.0/reference/mysql-compatibility.md]:
```suggestion
```
To fix CI.

**Comment by bb7133** [comment]
Please update the other versions.

**Comment by yikeke** [comment]
Please update the other versions.

**Pull request merged**